front/src/core-logic/adapters/Convention/HttpConventionGateway.ts

```diff
@@ -70,6 +73,25 @@ export class HttpConventionGateway implements ConventionGateway {
     );
   }
 
+  public getApiConsumersByconvention$(
+    params: WithConventionId,
+    jwt: InclusionConnectJwt,
+  ): Observable<ApiConsumerName[]> {
+    return from(
+      this.authenticatedHttpClient
+        .getApiConsumersByConvention({
+          urlParams: { conventionId: params.conventionId },
+          headers: { authorization: jwt },
+        })
+        .then((response) => {
+          return match(response)
+            .with({ status: 200 }, ({ body }) => body)
+            .with({ status: P.union(400, 401, 403, 404) }, logBodyAndThrow)
+            .otherwise(otherwiseThrow);
+        }),
```

Comment:
Il y a une gestion un peu meilleurs des 400 maintenant :

```suggestion
        .then((response) => match(response)
            .with({ status: 200 }, ({ body }) => body)
            .with({ status: 400 }, throwBadRequestWithExplicitMessage)
            .with({ status: P.union(401, 403, 404) }, logBodyAndThrow)
            .otherwise(otherwiseThrow);
        ),
```